graylog2-server/src/main/java/org/graylog2/alerts/Alert.java

```diff
@@ -16,18 +16,28 @@
  */
 package org.graylog2.alerts;
 
-import org.graylog2.plugin.database.Persisted;
 import org.joda.time.DateTime;
 
 import java.util.Map;
 
 /**
  * @author Dennis Oelkers <dennis@torch.sh>
  */
-public interface Alert extends Persisted {
+public interface Alert {
+    String getId();
```

Comment:
I don't see a reason that it must be part of the builder interface. It was left out on purpose, because the only part who should be responsible for creating new (not updating) new alerts should be the `AlertScanner` class.